src/arch/arm/64/head.S

```diff
@@ -23,6 +23,14 @@
 #define CR_ALIGN_CLEAR   0
 #endif
 
+#if defined(CONFIG_ARM_HYPERVISOR_SUPPORT) && defined(CONFIG_AARCH64_USER_CACHE_ENABLE)
+#define CR_USER_CACHE_OPS_SET (BIT(CONTROL_UCT) | BIT(CONTROL_UCI))
+#define CR_USER_CACHE_OPS_CLEAR 0
+#else
+#define CR_USER_CACHE_OPS_SET 0
+#define CR_USER_CACHE_OPS_CLEAR (BIT(CONTROL_UCT) | BIT(CONTROL_UCI))
```

Comment:
Is there are reason why `CONFIG_ARM_HYPERVISOR_SUPPORT` needs to be enabled in order to do the operations? From my understanding of the patch, the config option is on by default and so, regardless of whether the kernel is built as a hypervisor or not, user-level programs can execute the cache instructions? But it looks like the bits are not set unless the kernel is built as a hypervisor.

Reply:
(spotted by @lucypa)

Reply:
It seems like it should be `#if !defined(CONFIG_ARM_HYPERVISOR_SUPPORT) && defined(CONFIG_AARCH64_USER_CACHE_ENABLE)`. The bits are only present in the SCTLR_EL1 register and head.S programs SCTLR_EL2 when CONFIG_ARM_HYPERVISOR_SUPPORT is set.